### DISCUSSION TITLE
why pyinstaller never compile private module in project?

### DISCUSSION THREAD

**Discussion by natural-regular** [pyinstaller/pyinstaller#6454] [Help] [answered]
I wonder if issues system could help me with this.... ....

system info:
- macOS:Darwin iMac 21.1.0 Darwin Kernel Version 21.1.0: Wed Oct 13 17:33:23 PDT 2021; root:xnu-8019.41.5~1/RELEASE_X86_64 x86_64 i386 iMac19,1 Darwin
- Python:3.9.7
- pyinstaller:4.7

pyinstaller never compile module in project to dist

project struct:

project name is: testing
- `[testing]`
    - `[cmd]`
        + `__init__.py`
        + `Oracle.py`
    - `main.py`

`__init__.py`'s content is empty.

`Oracle.py`'s content

- def version(args):
     + print("v0.1")
- class OracleExec():
     + @staticmethod
     + def init(subparser):
         - print("---make oracle exec ---")
    
`main.py`'s content
- from cmd.Oracle import OracleExec
- if __name__ == '__main__':
    + OracleExec.init()

using `python3 main.py` , everything work fine.

but , run pyi under `testing` folder:

`pyinstaller main.py`

then calling compile target bin file:

`./dist/main/main`

errror message is:

> Traceback (most recent call last):
  File "main.py", line 12, in <module>
    from cmd.Oracle import OracleExec
ModuleNotFoundError: No module named 'cmd.Oracle'; 'cmd' is not a package
[37484] Failed to execute script 'main' due to unhandled exception!

files in `dist` folder:

`
Python  base_library.zip  lib-dynload  libcrypto.1.1.dylib  liblzma.5.dylib  libmpdec.3.dylib  libreadline.8.dylib  libssl.1.1.dylib  main
`

**Comment by bwoodsend** [comment]
Can you share your xref file (located at `build/main/xref-main.html`)? Python files are baked into the executable so there's no point looking for them in the dist folder - they won't be there.

**Comment by rokm** [comment, marked as answer]
Because `cmd` is a python system module, which gets collected instead of your package with the same name. Rename your `cmd` to something that doesn't clash with existing python modules, and it should work.

**Reply by natural-regular** [reply]
thank you, it works now.
